charmdir.go

```diff
@@ -425,9 +459,22 @@ func (dir *CharmDir) MaybeGenerateVersionString() (string, string, error) {
 		cmdArgs = []string{"bzr", "version-info"}
 	} else if _, err = os.Stat(filepath.Join(dir.Path, ".git")); err == nil {
 		cmdArgs = []string{"git", "describe", "--dirty", "--always"}
+	} else if file, err := os.Open(dir.join("version")); err == nil {
+		logger.Debugf("charm is not in version control, but uses a version file")
```

Comment:
I guess one thing to note is that if you move from version to scm or vice-versa you could end up with issues around tracking the version, but I guess they're probably holding it wrong, as there isn't a way to know if it once was with the prior revision control.

Reply:
I think one could add additional thoughts into it. 
But this would be outside of the scope of this PR.

Thanks for the review though!